Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+   *
+   * Set where clauses, linking them with the `AND` operator.
+   * Arguments can be passed in two forms. The simple one is useful for setting
+   * one condition, and consists of passing an argument for the field, one for
+   * the operator (can be omitted if '=') and one for the value.
+   * The other one consists of passing an array containing an array for every
+   * clause, each of which must contain two or three elements, in the same form
+   * as the first one.
+   * Every other way will raise an exception.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array|string Where clauses.
+   * @return self
+   *
+   * @throws \RuntimeException if the arguments aren't in the correct form.
+   */
+  public function where($clauses) {
+    try {
+      return $this->processWhere(func_get_args(), 'where');
+    } catch (Exception $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Set WHERE clauses.
+   *
+   * Set where clauses, linking them with the `OR` operator.
+   * The method behave like `Query::where()`.
+   * @see Query::where()
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array|string Where clauses.
+   * @return self
+   *
+   * @throws \RuntimeException if the arguments aren't in the correct form.
+   */
+  public function orWhere($clauses) {
+    try {
+      return $this->processWhere(func_get_args(), 'orWhere');
+    } catch (Exception $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Set WHERE clauses.
+   *
+   * Set clauses like: 'WHERE `field` IN (`value` AND `value` AND `value` ...)'.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $field Field name.
+   * @param array $values Values to check with.
+   * @return self
+   *
+   * @throws \RuntimeException if $value elements aren't in the correct form.
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if $field isn't a
+   * string.
+   */
+  public function whereIn($field, array $values) {
+    try {
+      return $this->processWhereIn($field, $values, 'whereIn');
+    } catch (Exception $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Set WHERE clauses.
+   *
+   * Set clauses like: 'WHERE `field` NOT IN (`value` AND `value` ...)'.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $field Field name.
+   * @param array $values Values to check with.
+   * @return self
+   *
+   * @throws \RuntimeException if $value elements aren't in the correct form.
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if $field isn't a
+   * string.
+   */
+  public function whereNotIn($field, array $values) {
+    try {
+      return $this->processWhereIn($field, $values, 'whereNotIn');
+    } catch (Exception $e) {
+      throw $e;
+    }
+  }
+
+  /**
+   * Check query result existence.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @return boolean Whether the result is set or not.
+   */
+  public function hasResult() {
+    return isset($this->result);
```

Comment:
`$this->result` is always set, since you declare it in the class. You could instead return
```php
!is_null($this->result)
```